packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -1842,15 +1842,16 @@ class RenderSliverOverlapAbsorber extends RenderSliver with RenderObjectWithChil
   }
 
   @override
-  bool hitTestChildren(SliverHitTestResult result, { required double mainAxisPosition, required double crossAxisPosition }) {
-    if (child != null) {
-      return child!.hitTest(
-        result,
-        mainAxisPosition: mainAxisPosition,
-        crossAxisPosition: crossAxisPosition,
-      );
-    }
-    return false;
+  bool hitTestChildren(
+    SliverHitTestResult result, {
+    required double mainAxisPosition,
+    required double crossAxisPosition,
+  }) {
+    return child?.hitTest(
+      result,
+      mainAxisPosition: mainAxisPosition,
+      crossAxisPosition: crossAxisPosition,
+    ) ?? false;
```

Comment:
I personally think the previous version is more readable and I'm not really for enforcing this with a linting rule.

Reply:
Neither one looks great, in my opinion. I can go ahead and revert this one 👍